operations/mimir/multi-zone.libsonnet

```diff
@@ -26,6 +26,8 @@
     multi_zone_store_gateway_replicas: 0,
     multi_zone_store_gateway_max_unavailable: 10,
 
+    multi_zone_store_gateway_unregister_on_shutdown: false,
```

Comment:
I would keep the same consistent with `unregister_ingesters_on_shutdown`, so `unregister_store_gateways_on_shutdown`. This feature is unrelated from "multi-zone", so no need to prefix it with `multi_zone_`.

Reply:
After offline discussion, the challenge here is that multi-zone needs to be specifically set to something other than the default for `store-gateway.sharding-ring.unregister-on-shutdown`. This would be a regression requiring any operators using multi-zone to set `unregister_store_gateways_on_shutdown=false` manually, or otherwise still require an extra variable to control.

Reply:
Discussed on Slack. Actually we need a different setting whether it's multi-zone or not, so let's move on with this approach.